azure-boards-common/src/main/java/com/synopsys/integration/azure/boards/common/oauth/AzureCredential.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * azure-boards-common
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.azure.boards.common.oauth;
+
+import java.io.IOException;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.auth.oauth2.RefreshTokenRequest;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.http.GenericUrl;
+
+public class AzureCredential extends Credential {
+    private String clientSecret;
+    private String redirectUri;
+    private String cachedRefreshToken;
+
+    public AzureCredential(AccessMethod method) {
+        super(method);
+    }
+
+    public AzureCredential(Builder builder) {
+        super(builder);
+        this.clientSecret = builder.getClientSecret();
+        this.redirectUri = builder.getRedirectUri();
+        this.cachedRefreshToken = builder.getCachedRefreshToken();
+    }
+
+    public String getClientSecret() {
+        return clientSecret;
+    }
+
+    public String getCachedRefreshToken() {
+        return cachedRefreshToken;
+    }
+
+    public String getRedirectUri() {
+        return redirectUri;
+    }
+
+    @Override
+    protected TokenResponse executeRefreshToken() throws IOException {
+        String refreshToken = getRefreshToken();
+        if (refreshToken == null) {
```

Comment:
getRefreshToken will return null by default if the credential isn't set.  If getRefreshToken is null check the cachedRefreshToken to make sure it isn't null.  If the cached token that was passed to the build is null return a null response.  This has the same behavior of the base class when the refreshToken member is null.